python/cudf_polars/tests/experimental/test_stats.py

```diff
@@ -407,3 +410,81 @@ def test_base_stats_join_key_info(engine):
         == implied_unique_count
     )
+
+    # Check basic collect_statistics behavior
+    stats = collect_statistics(ir, config_options)
+    local_unique_count = stats.column_stats[ir]["cust_id"].unique_count.value
+    source_unique_count = (
+        stats.column_stats[ir]["cust_id"].source_info.unique_stats().count.value
+    )
+    assert local_unique_count == source_unique_count
+    assert stats.row_count[ir].value == q.collect().height
+
+
+@pytest.mark.parametrize("io_partitioning", [True, False])
+@pytest.mark.parametrize("reduction_planning", [True, False])
+@pytest.mark.parametrize("use_join_heuristics", [True, False])
+@pytest.mark.parametrize("use_sampling", [True, False])
+@pytest.mark.parametrize("default_selectivity", [0.5, 1.0])
+@pytest.mark.parametrize("kind", ["parquet", "csv", "frame"])
+def test_stats_planning_options(
+    tmp_path,
+    kind,
+    io_partitioning,
+    reduction_planning,
+    use_join_heuristics,
+    use_sampling,
+    default_selectivity,
+):
+    # Create temporary GPU Engine
+    engine = pl.GPUEngine(
+        raise_on_fail=True,
+        executor="streaming",
+        executor_options={
+            "scheduler": DEFAULT_SCHEDULER,
+            "shuffle_method": "tasks",
+            "target_partition_size": 10_000,
+            "max_rows_per_partition": 1_000,
+            "stats_planning_options": {
+                "io_partitioning": io_partitioning,
+                "reduction_planning": reduction_planning,
+                "use_join_heuristics": use_join_heuristics,
+                "use_sampling": use_sampling,
+                "default_selectivity": default_selectivity,
+            },
+        },
+    )
+
+    # Define "complicated" query that uses all stats planning options
+    sales = pl.DataFrame(
+        {
+            "order_id": [1, 2, 3, 4, 5, 6],
+            "customer_id": [101, 102, 101, 103, 102, 101],
+            "amount": [50.0, 75.0, 30.0, 120.0, 85.0, 40.0],
+            "product": ["A", "B", "A", "C", "B", "A"],
+        }
+    )
+    sales = make_lazy_frame(sales, kind, path=tmp_path / f"sales_{kind}")
+    customers = pl.DataFrame(
+        {
+            "customer_id": [101, 102, 103],
+            "customer_name": ["Alice", "Bob", "Charlie"],
+            "region": ["North", "South", "North"],
+        }
+    )
+    customers = make_lazy_frame(customers, kind, path=tmp_path / f"customers_{kind}")
+    q_join = sales.filter(pl.col("amount") < 100.0).join(
+        customers, on="customer_id", how="inner"
+    )
+    q_gb = q_join.group_by("customer_id").agg(
+        [
+            pl.col("amount").sum().alias("total_amount"),
+            pl.col("order_id").count().alias("order_count"),
+            pl.col("customer_name").first().alias("name"),
+            pl.col("region").first().alias("region"),
+        ]
+    )
+    q = q_gb.sort("customer_id")
+
+    # Verify the query runs correctly
+    assert_gpu_result_equal(q, engine=engine)
```

Comment:
What guarantees do we have that this actually did exercise all the choices made by the various statistics options?

I think if we have other tests that actually get the IR node, collects the statistics, and makes assertions about them, then having this just as a basic correctness test is good. If we don't have those tests, then perhaps we could add them or update this test to collect the stats and make some assertions, depending on which options are enabled.